pyanaconda/dbus/structure.py

```diff
@@ -273,23 +273,21 @@ def generate_string_from_data(obj, skip=None, add=None):
 
     :param obj: a data object
     :param skip: a list of names that should be skipped or None
-    :param add: a list of strings to add or None
+    :param add: a dictionary of attributes to add or None
     :return: a string representation of the data object
     """
-    attributes = []
-    skipped_names = set(skip) if skip else set()
-    extra_attributes = list(add) if add else []
+    dictionary = generate_dictionary_from_data(obj)
 
-    for field in get_fields(obj).values():
-        if field.data_name in skipped_names:
-            continue
+    for name in skip or list():
+        dictionary.pop(name, None)
 
-        attribute = "{}={}".format(field.data_name, repr(field.get_data(obj)))
-        attributes.append(attribute)
+    for name in add or dict():
```

Comment:
I don't like this construct but it works so if you want, you can leave it this way.